examples/resources/nomad_with_modules/main.tf

```diff
@@ -0,0 +1,95 @@
+module "nomad" {
+  source          = "/home/eslam/work/currently/terraform-provider-grid/modules/nomad-module"
+  ssh_key         = local.ssh_key
+  first_server_ip = local.first_server_ip
+  network         = local.network
+  servers         = local.servers
+  clients         = local.clients
+}
+
+locals {
+  ssh_key         = file("~/.ssh/id_rsa.pub")
+  first_server_ip = "10.1.2.2"
+
+  network = {
+    nodes       = [29]
+    ip_range    = "10.1.0.0/16"
+    name        = "nomadtest"
+    description = "new network for nomad"
+  }
+
+  servers = [
+    {
+      name        = "server1"
+      node        = 29
+      cpu         = 2
+      memory      = 1024
+      publicip    = false
+      publicip6   = false
+      planetary   = true
+      mount_point = "/mnt"
+      disk = {
+        name = "server1dsk"
+        size = 5
+      }
+    },
+    {
+      name        = "server2"
+      node        = 29
+      cpu         = 2
+      memory      = 1024
+      publicip    = false
+      publicip6   = false
+      planetary   = true
+      mount_point = "/mnt"
+      disk = {
+        name = "server2dsk"
+        size = 5
+      }
```

Comment:
also, a lot of the variables here are repeated. a map with for_each can make it simpler